### DISCUSSION TITLE
Default password hashing algorithm

### DISCUSSION THREAD

**Discussion by TomaszGasior** [symfony/symfony#42762] [Q&A]
What is the default password hashing algorithm when set to `auto` in `security.yaml` in current Symfony versions? As I understand from documentation and from the code, in 5.3 that's 'bcrypt'. As I remember, in previous versions 'argon' was used by default. Why this change happened? Or maybe I remember something wrong?

**Comment by ahundiak** [comment]
Speculating a bit here but I believe auto now uses [PASSWORD_DEFAULT](https://www.php.net/manual/en/function.password-hash.php) which currently means bcrypt.  ARGON2I and ARGON2ID need to be compiled into PHP and thus are not always available.  

As you pointed out, [Symfony 5.3 documentation](https://symfony.com/doc/current/security.html#c-hashing-passwords) indicates that bcrypt is the 'best' possible hashing algorithm.  In 4.4 sodium (aka argon) was used.

**Reply by TomaszGasior** [reply]
Yep, that's mostly what I know. :) Basically I want to get to know why this change happened, why argon was changed back to bcrypt. 

As I understand, Sf 5.3 uses new password hasher component which explicitly defaults to bcrypt when auto:
- https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/PasswordHasher/Hasher/PasswordHasherFactory.php#L121
- https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/PasswordHasher/Hasher/NativePasswordHasher.php#L28
